src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -58,6 +60,8 @@ internal partial class CurlHandler : HttpMessageHandler
         private DecompressionMethods _automaticDecompression = DecompressionMethods.GZip | DecompressionMethods.Deflate;
         private SafeCurlMultiHandle _multiHandle;
         private GCHandle _multiHandlePtr = new GCHandle();
+        private bool _preAuthenticate = false;
+        private readonly CredentialCache _credentialCache = new CredentialCache();
```

Comment:
Should the credential cache be lazily allocated only when needed?  Or is it always needed?